src/components/LeftNavigation/NavCategory/NavCategory.tsx

```diff
@@ -77,15 +78,24 @@ export const NavCategory = ({
   const ctx = useContext(NavigationContext);
 
   useEffect(() => {
-    const itemStyle = getNavBarClassNames(isExpanded, isActive, disabled);
+    const active = childIsActive;
+    const itemStyle = getNavBarClassNames(isExpanded, active, disabled);
     setNavItemClass(itemStyle);
-  }, [isExpanded, isActive, disabled]);
+  }, [isExpanded, isActive, disabled, childIsActive]);
 
   useEffect(() => {
     const iconStyles = getIconClass(icon);
     setIconClass(iconStyles);
   }, [icon]);
 
+  useEffect(() => {
+    let hasActiveLinks = false;
+    Children.map(children, (child) => {
+      hasActiveLinks = hasActiveLinks || child.props.href === ctx.currentUrl;
+    });
+    setChildIsActive(hasActiveLinks);
```

Comment:
You don't want to use `map` because it won't end early. What you want is [the array method "some"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). 
```suggestion
    const hasActiveLinks = Children.some(children, (child) => child.props.href === ctx.currentUrl);
    setChildIsActive(hasActiveLinks);
```

Reply:
yes, I tried using some() but the Children helper does not support it :)

Reply:
Huh, I assumed that `Children` was an array, but that is not the case. NVM, yours is the most performant solution that I'm able to think of. Good call 👍

Reply:
Yeah, I thought it was an array too but it's some kind of helper React object.